### PR TITLE
[stable/traefik]Added loadBalancerIP too the traefik service template

### PR DESCRIPTION
**Pull request by vongohren**
This can be set through a value called googleStaticIP, which is None by default. This was needed to make it possible to set up your loadbalancer with helm and assigning an already reserverd static ip in GCP. Im not sure if this works on other clouds, therefor have I called it googleStaticIP. Updated README with proper values

### CODE REVIEW

**Comment by krancour** [comment]
I think if we start introducing cloud-specific functionality to this chart, those features should maybe be cordoned off in their own section of `values.yaml`. i.e. Perhaps add a `gcp` section and move this option under there. Additionally, it might be worthwhile to introduce another option that is undefined by default, but can be set to indicate what cloud the chart is being deployed to. This way, if (as?) cloud-specific options continue to grow, it will be easy to write templates that quickly determine which swath of cloud-specific options are applicable.

Also, you'd need to bump the chart's minor version number in `chart.yam`.

**Comment by vongohren** [comment]
@krancour I did a commit now which made it more cloud specific. But would you like to elborate on the last part? Not sure if I get how it can be done?

My thought would be to have a value saying `cloud`

If set to f.ex "gcp", how would that be used?
You thinking of then having templates with if statements based on the value of cloud? Then those options would be available, and if no cloud specified, those would not be available?

File: stable/traefik/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: traefik
-version: 1.7.2
+version: 1.7.3
```
Comment (krancour) [inline stable/traefik/Chart.yaml]: This PR adds a feature; not a bug fix. So, this should be 1.8; not 1.7.3.

**Comment by krancour** [comment]
> You thinking of then having templates with if statements based on the value of cloud? Then those options would be available, and if no cloud specified, those would not be available?

Yes. Exactly. So the following:

```
{{- if .Values.gcp.staticIP }}
loadBalancerIP: {{ .Values.gcp.staticIP }}
{{- end }}
```

Should change to something like:

```
{{- if and (eq .Values.cloud "gcp") (.Values.gcp.staticIP) }}
loadBalancerIP: {{ .Values.gcp.staticIP }}
{{- end }}
```

These sort of guards can be used to ensure that logic that's only applicable to a given cloud isn't applied to deployments to some other cloud.

File: stable/traefik/README.md [hunk 2]
```diff
@@ -119,6 +119,7 @@ The following tables lists the configurable parameters of the Traefik chart and
 | `accessLogs.enabled`            | Whether to enable Traefik's access logs                              | `false`                                   |
 | `accessLogs.filePath`           | The path to the log file. Logs to stdout if omitted                  | None                                      |
 | `accessLogs.format`             | What format the log entries should be in. Either `common` or `json`  | `common`                                  |
+| `gcp.staticIP`            | An available IP you have reserved on Google Cloud Platform           | None                                      |
```
Comment (unguiculus) [inline stable/traefik/README.md]: Don't call it `gcp.staticIP`. This is not GCP-specific. Just call it `loadBalancerIP`. Please format the table nicely.

File: stable/traefik/Chart.yaml (hunk 1 above)
Comment (unguiculus) [reply stable/traefik/Chart.yaml]: I agree.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/traefik/README.md (hunk 2 above)
Comment (vongohren) [reply stable/traefik/README.md]: What do you think of @krancour's suggestion? As I mentioned in my comment im not sure how this works with other clouds. But you know that this works with more than gcp?

**Comment by vongohren** [comment]
I think that its a good idea @krancour, but now @unguiculus came with a change request meaning that it probably works with multiple clouds. So the cloud specifics have to come with another PR i think. I did a last commit with loadBalancerIP, you guys should agree on what to do.

**Comment by krancour** [comment]
@vongohren you mentioned not knowing if/how this works in clouds other than GCP, and I have to admit, I let that fool me into thinking there was _probably_ something cloud-specific happening here. Especially in light of @unguiculus comment, I've reconsidered that. I think that setting the `loadBalancerIP` in the service is always going to be ok and if a cloud provider support it, it supports it. Or it may not. It will all sort itself out. So I agree with @unguiculus.

Having said that, this setting probably isn't equally applicable to all _types_ of services. e.g. It makes sense when the service is one of `type: LoadBalancer`, but if it's one of `type: NodePort` (for instance), it may not make as much sense.

So, should the service type be considered when deciding to set or not set this attribute?

**Comment by vongohren** [comment]
Yeah @krancour, something like this? Im quite new to this, but it sounds logical. But either you or @unguiculus have to answer if this is the case, and that you want this change

```
{{- if and (eq .Values.serviceType "LoadBalancer") (.Values.loadBalancerIP) }}
  loadBalancerIP: {{ .Values.loadBalancerIP }}
{{- end }}
```

**Comment by unguiculus** [comment]
You can check the chart examples for reference: https://github.com/kubernetes/helm/blob/master/docs/examples/nginx/templates/service.yaml

**Comment by vongohren** [comment]
Looks like what im doing here, so I would say its good to go?

**Review by unguiculus — APPROVED** [review]

**Comment by unguiculus** [comment]
Bump the chart version as suggested and this is good to go.

**Comment by vongohren** [comment]
@unguiculus done!

Guess i missed that one since he commented something else earlier

**Comment by krancour** [comment]
@vongohren the version still doesn't appear changed.

**Comment by vongohren** [comment]
@krancour ah no sorry, I assumed 1.8 was latest, but someone captured that version, so I now do 1.9. Now its done.

**Comment by krancour** [comment]
LGTM

**Pull request merged**